### DISCUSSION TITLE
Lint file with $ in its name

### DISCUSSION THREAD

**Discussion by thefrana** [eslint/eslint#14254] [Q&A]
Hi,
when I try `eslint path/to/file/$something.foo` it tries to resolve `$something` as an variable (I'm on Linux) and it becomes only `path/to/file/.foo` since variable does not exist. Is there any way how to resolve this issue? Specifically I'm using ESLint with `lint-staged`. The latter utility run command on every file and I cannot properly escape the filename. Right now I am exporting variable with `export something='$something'`, but that is far from ideal.

Thank you

**Comment by nzakas** [comment]
That’s part of your shell expansion. You might want to post this question to lint-staged project.

**Reply by thefrana** [reply]
Yeah, I know its a shell issue. I asked here someobody does have some workaround, but I will also ask `lint-staged` project. Thanks

**Reply by aladdin-add** [reply]
have you tried `eslint "path/to/file/$something.foo"` - to avoid shell expansion?
